channel.go

```diff
@@ -281,6 +282,7 @@ func NewChannel(serviceName string, opts *ChannelOptions) (*Channel, error) {
 	}
 	ch.mutable.state = ChannelClient
 	ch.mutable.conns = make(map[uint32]*Connection)
+	ch.mutable.closed = make(chan struct{})
```

Comment:
why is `closed` mutable?  I think it's immutable (set on channel creation) and should only ever be closed once

Reply:
Makes sense.  Moved it out of `mutable`.